### ISSUE TITLE
[BUG]: ifx/ifort Fortran compiler not found in IDE mode with CMake

### ISSUE THREAD

**Issue by kc9jud** [compiler-explorer/compiler-explorer#5689]
### Describe the bug

CMake cannot find the Fortran compiler when compiling with ICX or ICC.

### Steps to reproduce

1. Visit godbolt.org, create new IDE window.
2. Create mixed C++/Fortran project with appropriate `CMakeLists.txt`.
3. Try to compile with GCC: works.
4. Try to compile with icx or icc: 
```
CMake Error at CMakeLists.txt:1 (project):
  No CMAKE_Fortran_COMPILER could be found.

  Tell CMake where to find the compiler by setting either the environment
  variable "FC" or the CMake cache entry CMAKE_Fortran_COMPILER to the full
  path to the compiler, or to the compiler name if it is in the PATH.
```

### Expected behavior

The Fortran compiler ifx/ifort should be found when using icx/icc, respectively.

### Reproduction link

https://godbolt.org/z/9WcxPjETr

### Screenshots

Not applicable

### Operating System

Not applicable

### Browser version

Not applicable

**Comment by partouf** [comment]
Ah yes, thank you for the example. I believe this could work if you select Fortran as the language + compiler, but then you can't add fmt as a library.

Currently the way we set things up is a bit messy, but if we continue that messyness, we could just add the fortran variables here https://github.com/compiler-explorer/compiler-explorer/blob/main/lib/base-compiler.ts#L319 and point to the expected compiler based on the c/c++ compiler's path.

**Comment by kc9jud** [comment]
Yeah, I think that would be a decent fix.

Ideally, one would be able to control the compilers for different languages independently, in order to play with things like binding compatibility between compilers. This would also be useful if, e.g., the C++ compiler doesn't have an associated Fortran compiler (such as clang, notwithstanding the flang chaos).

**Comment by partouf** [comment]
Our intel fortran compilers aren't actually installed in the same directory or versioned the same as the intel c/c++ compilers. So we can't really assume the location of the fortran compilers in this case.

We would need a mechanism to couple certain versions together, and I'm not really a fan of such a coupling.

Not sure how to procede

**Comment by kc9jud** [comment]
I agree, coupling versions together seems wrong, especially since one might want to test linking between different versions of compilers.

It seems like the "right" way is to specify the different compilers independently... which will require more design, especially from a UI/UX perspective.

**Comment by partouf** [comment]
Right, perhaps the best way would be to be have a Compiler Overrides type of deal where you can select any kind of compiler like how the GCC Toolchain selection works for Clang.

We do need a haul-over for CMake either way, because we'd like to have more build systems to accommodate other languages, but that's more of a plan for the far future.
